### PR TITLE
[CDAP-17466] Authentication functionality for CDAP on Kubernetes setup

### PR DESCRIPTION
**Pull request by edvinas-maciulis**
Design document: https://wiki.cask.co/display/CE/Authentication+feature+for+CDAP+on+Kubernetes+deployment
JIRA Ticket: https://cdap.atlassian.net/browse/CDAP-17466

### CODE REVIEW

**Comment by edvinas-maciulis** [comment]
@googlebot I signed it!

**Comment by edvinas-maciulis** [comment]
@googlebot I signed it!

**Comment by edvinas-maciulis** [comment]
@googlebot I signed it!

**Comment by edvinas-maciulis** [comment]
@chtyim - when you have a chance, could you take a look, please.

**Comment by edvinas-maciulis** [comment]
@chtyim - how does it look for you?

File: controllers/constants.go [hunk 1]
```diff
@@ -50,7 +54,7 @@ const (
 	confUserInterfaceBindPort = "dashboard.bind.port"
 
 	// default values
-	defaultImage             = "gcr.io/cdapio/cdap:latest"
+	defaultImage             = "cdap_google:0.0.4"
```
Comment (chtyim) [inline controllers/constants.go]: Please revert this change.

File: config/samples/cdap_v1alpha1_cdapmaster.yaml [hunk 2]
```diff
@@ -14,3 +14,7 @@ spec:
     metadata.storage.implementation: elastic
     metadata.elasticsearch.cluster.hosts: elasticsearch-master
     hdfs.user: root
+    security.enabled: "true"
+    security.auth.server.announce.address: ""
```
Comment (chtyim) [inline config/samples/cdap_v1alpha1_cdapmaster.yaml]: Should use `security.auth.server.announce.urls` instead. The announce.address was deprecated a long while ago.

File: controllers/testdata/cdap_master_cr.json [hunk 3]
```diff
@@ -43,7 +43,9 @@
       "metadata.elasticsearch.cluster.hosts": "elasticsearch-master",
       "metadata.storage.implementation": "elastic",
       "router.bind.port": "11015",
-      "router.server.address": "cdap-test-router"
+      "router.server.address": "cdap-test-router",
+      "security.authentication.handlerClassName":	"io.cdap.cdap.security.server.BasicAuthenticationHandler",
```
Comment (chtyim) [inline controllers/testdata/cdap_master_cr.json]: Only one space is needed after the `:`.

**Comment by chtyim** [comment]
Also, running of `make test` has failure. Can you please fix it as well?

**Comment by chtyim** [comment]
Changes are incorporated in #61

**Comment by edvinas-maciulis** [comment]
> Also, running of `make test` has failure. Can you please fix it as well?

That's done - thanks @chtyim for pointing out.

**Comment by chtyim** [comment]
The changes has been incorporated in #61. Closing this PR.
